### DISCUSSION TITLE
Swellbox Modeling

### DISCUSSION THREAD

**Discussion by mutti-wilson** [GrandOrgue/grandorgue#922] [General]
I was reading through @larspalo blog post about Swellbox modeling and I was wondering if anything had come of it? @larspalo you had mentioned using a shelf filter as an option. Would that be a high and low shelf filter? You wrote "Thus any kind of filter that allows the bass frequencies to be less reduced than the middle/high will be a better model than pure amplitude reduction." which seems spot on. It seems as though if you could reduce dB's across the frequency scale with the highs frequencies being reduced more than the lows that would be close enough to the real thing.

Ideally there would be a tab in the sound settings where EQ and this could be adjusted since all sample sets may not react the same way. If a profile could be saved and either manually changed or loaded when a specific sample set is played that would be even better. I know this may be asking for too much but I think it is one of the bigger sound quality areas that GO is lacking in. I have more ideas than skills as far as programming but I would be willing to help test.

**Comment by mutti-wilson** [comment]
This was also a very interesting read.

https://sourceforge.net/p/ourorgan/mailman/ourorgan-developers/thread/CE0FCD66-BCE4-462C-ADC7-ED9D9EFC083A@gmail.com/?page=0

What was the "old" swellbox modeling compared to the new one that was mentioned? Was your patch ever implemented?

**Comment by larspalo** [comment]
I had a working patch made in august 2020 that implemented a filter model on the windchest level (not per sample, which would allow for much greater diversity and greater tweaking possibilities). The patch never made its way into the official repository because of lack of feedback from both users and developers. It's still available at https://sourceforge.net/p/ourorgan/mailman/message/37081601/ but the patch will likely need some re-working as many things have changed in the code since then, but in essence it should still be usable with a manual application of the code changes.

My thoughts are that this on a per pipe sample level would be amazing, and it's definitely on the todo list (an issue is already created for it), but I'm presently working on a complete odf creation tool that will eventually make all manual odf edits a thing of the past! Thus, I'm not going to start the work on the filtering approach to swellbox modeling for (quite) a while.

**Reply by mutti-wilson** [reply]
A tool for ODF creation/modification sounds great. Looking forward to seeing what you come up with.

Is the current swell box model amplitude reduction evenly across all frequencies?

BTW have you heard of Markus Wargh? He's the organist at Luleå Domkyrka.

**Reply by larspalo** [reply]
Current GO swellbox modeling is amplitude only, no filtering.

Yes, I know him pretty well both privately and professionally. I actually had him as a teacher for one out of four years of my Organist (Master of fine arts in Church music) study time.

**Reply by mutti-wilson** [reply]
That's what I figured on the swell box.

That's great. He is affiliated with my congregation's denomination here in the US. My wife is from Alta, Norway and grew up listening to him and going to sangaften services from time to time. He played a concert/song service in my church a couple years ago. I keep in touch with him here and there. Small world.

**Comment by hnb2907** [comment]
> I had a working patch made in august 2020 that implemented a filter model on the windchest level (not per sample, which would allow for much greater diversity and greater tweaking possibilities). The patch never made its way into the official repository because of lack of feedback from both users and developers. It's still available at https://sourceforge.net/p/ourorgan/mailman/message/37081601/ but the patch will likely need some re-working as many things have changed in the code since then, but in essence it should still be usable with a manual application of the code changes.
> 

Hi @larspalo,

This is something I would be interested to try out.  I notice that the patch did not make it into the main code, and modifying it to work again is probably beyond my capabilities.
Is this something that is still on the to-do list for GO, and is it easy to apply the patch (what are the full steps to do this and compile it for ubuntu)?


Best wishes,
Chris.

**Reply by larspalo** [reply]
@hnb2907 Yes, this is still TODO. Most likely we'll eventually try to implement some eq functionality that could be usable for both pipe voicing and swellbox modeling.

The old patch would certainly need to be modified as many things have changed since it was written, but moreover - as said above - I think that the approach to the eq should be somewhat changed from how it was done in that patch. The resulting sound from the shelf filter (that Nick Appleton actually designed, and I implemented) was pretty good and not quite as "destructive" as the simple lowpass filter was.
